ee/clickhouse/queries/funnels/test/test_funnel_trends.py

```diff
@@ -970,3 +930,309 @@ def test_one_person_in_multiple_periods_and_windows_in_strict_funnel(self):
+            distinct_id="user_one",
+            team=self.team,
+            timestamp="2021-05-05 00:00:00",
+            properties={"$browser": "Chrome"},
+        )
+
+        _create_person(distinct_ids=["user_two"], team=self.team)
+        _create_event(
+            event="step one",
+            distinct_id="user_two",
+            team=self.team,
+            timestamp="2021-05-02 00:00:00",
+            properties={"$browser": "Chrome"},
+        )
+        _create_event(
+            event="step two",
+            distinct_id="user_two",
+            team=self.team,
+            timestamp="2021-05-03 00:00:00",
+            properties={"$browser": "Chrome"},
+        )
+        _create_event(
+            event="step three",
+            distinct_id="user_two",
+            team=self.team,
+            timestamp="2021-05-05 00:00:00",
+            properties={"$browser": "Chrome"},
+        )
+
+        _create_person(distinct_ids=["user_three"], team=self.team)
+        _create_event(
+            event="step one",
+            distinct_id="user_three",
+            team=self.team,
+            timestamp="2021-05-03 00:00:00",
+            properties={"$browser": "Safari"},
+        )
+        _create_event(
+            event="step two",
+            distinct_id="user_three",
+            team=self.team,
+            timestamp="2021-05-04 00:00:00",
+            properties={"$browser": "Safari"},
+        )
+        _create_event(
+            event="step three",
+            distinct_id="user_three",
+            team=self.team,
+            timestamp="2021-05-05 00:00:00",
+            properties={"$browser": "Safari"},
+        )
+
+        filter = Filter(
+            data={
+                "insight": INSIGHT_FUNNELS,
+                "display": TRENDS_LINEAR,
+                "interval": "day",
+                "date_from": "2021-05-01 00:00:00",
+                "date_to": "2021-05-13 23:59:59",
+                "funnel_window_days": 7,
+                "events": [
+                    {"id": "step one", "order": 0},
+                    {"id": "step two", "order": 1},
+                    {"id": "step three", "order": 2},
+                ],
+                "breakdown_type": "event",
+                "breakdown": "$browser",
+            }
+        )
+        funnel_trends = ClickhouseFunnelTrends(filter, self.team, ClickhouseFunnel)
+        result = funnel_trends.run()
+
+        self.assertEqual(len(result), 2)
+
+        for res in result:
+            if res["breakdown_value"] == "Chrome":
+                self.assertEqual(res["data"], [100.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+            elif res["breakdown_value"] == "Safari":
+                self.assertEqual(res["data"], [0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+            else:
+                self.fail(msg="Invalid breakdown value")
+
+    def test_funnel_step_breakdown_person(self):
```

Comment:
Can we also have a test where we have another person with `properties={"$browser": "rando"}` ?

If I understand correctly, this will be included in `breakdown_values`, which means it will be in the right outer join, and should be all empty?

Reply:
i'm not following. where's there a right outer join for funnel trensd?

Reply:
Oops, *full outer join* - (right was recently added in time to convert, lol). But point still stands?

Reply:
Sidenote: We should make it a `RIGHT OUTER JOIN` here as well? Since we shouldn't have anything outside of that range

Reply:
technically yes I think. let's make that change in a different PR. As for the extra person with different property, theyll only be included if they did some related event within the time period being queried. If i just make another person with the property it won't register